homeassistant/components/onewire/sensor.py

```diff
@@ -124,35 +125,41 @@ def hb_info_from_type(dev_type="std"):
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Old way of setting up 1-Wire platform."""
+    adjusted_config = config
```

Comment:
We should import the config to a config flow here.

Reply:
Happy to implement - is there a good example of this?

Reply:
Here's one example:
https://github.com/home-assistant/core/blob/9ee97cb213d659aa9b6149484c0a42522decba78/homeassistant/components/nut/sensor.py#L60-L67

Reply:
I have it working on my machine, but that means that I have to completely drop support for OWFS, and force-migrate those to OWServer which I was trying to avoid.
I'm also not sure what to do about "CONF_NAMES" which was used to specify friendly names for devices (and all associated sensors). 

For example, having the following configuration in the YAML would generate the sensor entity_id `"sensor.my_ds18b20_temperature"` instead of `"sensor.10_111111111111_temperature"`
```
"names": {
            "10-111111111111": "My DS18B20",
        }
```

Is there a way to avoid those two breaking changes?

Reply:
I don't see any reason you cannot import the name mapping, and use it when naming the created devices/entities, despite not supporting setting it via the UI. I'm not aware of any hard rule that requires everything in a config entry to be ui configurable. Sure normally that is desirable, but in some compatibility cases like this, I'm not sure it is necessary. Indeed I'd be slightly surprised if some other integration had not done something similar for compatibility reasons.

As long as these set device/entity names eventually get stored in the same way as if the user had changed them via the UI, then things basically just work from the user perspective.

Of course, said part of the entry should be clearly documented as temporary transition code, to avoid breaking backwards compatibility.  Later when yaml import is removed, a config entry migration can drop the mappings from the config entry, and from that point forward, name customization can only be done via the ui/api. 

I cannot comment on OWFS, beyond mentioning that I see no reason why somebody could not create some form of `pyowfs` or `owfs-ha` library based on the existing code, and update it to use that, which brings it back into compliance with the library policy, and would allow it to be configured via config flow. If that was done first, it would eliminate this whole concern.

Reply:
I think it would be ok to import also the OWFS config and just forward that as an entry to the platform, including a name config item. We don't do any connection checks for it and make a comment about this case.

We also need to handle a restart and re-import of an OWFS config to an existing OWFS config entry. Ie, if the user has changed a config item for OWFS we should update the existing corresponding config entry. Probably keep track of the mound directory for each config entry, and use that to map them.

Reply:
New commits f855383 / a54662affe8f27c46af02982baec42229648de68 hopefully implements this correctly.